tests/unit/validation/test_validation_functions.py

```diff
@@ -431,6 +433,31 @@ def test_validate_values__infinite_sigmas(sensor_type, parameter):
         assert not isinstance(error, InfinityError)
 
 
+@pytest.mark.parametrize(
+    ("sensor_type", "parameter"),
+    [
+        ("sym_power_sensor", ["p_sigma", "q_sigma"]),
+        ("asym_power_sensor", ["p_sigma", "q_sigma"]),
+    ],
+)
+def test_validate_values__bad_p_q_sigma(sensor_type, parameter):
+    def arbitrary_fill(array, sensor_type, parameter):
+        if sensor_type == "sym_power_sensor":
+            array[parameter[0]] = 0.5
+        else:
+            array[parameter[1]][0][0] = 0.6
+            array[parameter[0]][0][1] = 0.7
+            array[parameter[1]][0][2] = 0.8
```

Comment:
please also add bad-weather scenarios, e.g.:

* single invalid `sym_power_sensor` with the `p_sigma` is nan but `q_sigma` is set
* single invalid `sym_power_sensor` with the `p_sigma` is set but `q_sigma` is nan
* 1 valid and one invalid `sym_power_sensor` (only the invalid one should be flagged)
* single invalid `asym_power_sensor` with all `p_sigma` are nan but all `q_sigma` are set
* single invalid `asym_power_sensor` with all `p_sigma` are set but all `q_sigma` are nan
* single invalid `asym_power_sensor` with both `p_sigma` and `q_sigma` set for some phases but nan for others (e.g.: `{"p_sigma" = [nan, 1.0, nan], "q_sigma = [nan, 1.0, nan]}`)
* single invalid `asym_power_sensor` with some combination of set and nan `p_sigma` and `q_sigma` (e.g.: `{"p_sigma" = [nan, 1.0, nan], "q_sigma = [1.0, nan, nan]}`)
* single invalid `asym_power_sensor` with all `p_sigma` are set but all `q_sigma` are nan
* 1 valid and one invalid `asym_power_sensor` as before

of course, both all set **and** all nan should be checked as well

I know it's a lot of test cases but it's a lot of edge cases as well and we need to catch them all